examples/main.cpp

```diff
@@ -0,0 +1,156 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "ProtectedCacheExample.h"
+#include "ReadExample.h"
+#include "WriteExample.h"
+
+#include <getopt.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <iostream>
+
+enum RunTest : int {
+  read_example = 0b1,
+  write_example = 0b10,
+  cache_example = 0b100,
+  all_examples = read_example | write_example | cache_example
+};
+
+constexpr auto usage =
+    "usuage is \n -a [--all] : run all examples \n -e [--example] : run "
+    "example [read, write, cache] \n -i [--key_id] here.access.key.id \n -s "
+    "[--key_secret] here.access.key.secret \n"
+    " -c [--catalog] the HRN of the catalog. \n"
+    " -l [--layer_id] the of the layer inside the catalog to which you want to "
+    "publish data(required for write example) \n"
+    " -h [--help]: show usage \n For "
+    "instructions on how to get the access key ID and access key secret, see "
+    "the [Get "
+    "Credentials](https://developer.here.com/olp/documentation/access-control/"
+    "user-guide/topics/get-credentials.html) section in the Terms and "
+    "Permissions User Guide.";
+
+int RunExamples(const AccessKey& access_key, int tests_to_run,
+                const std::string& catalog, const std::string& layer_id) {
+  if (tests_to_run & RunTest::read_example) {
+    std::cout << "Read Example" << std::endl;
+    if (RunExampleRead(access_key, catalog)) {
+      std::cout << "Read Example failed" << std::endl;
+      return -1;
+    }
+  }
+
+  if (tests_to_run & RunTest::write_example) {
+    std::cout << "Write example" << std::endl;
+    if (RunExampleWrite(access_key, catalog, layer_id)) {
+      std::cout << "Write Example failed" << std::endl;
+      return -1;
+    }
+  }
+
+  if (tests_to_run & RunTest::cache_example) {
+    std::cout << "Protected cache example" << std::endl;
+    if (RunExampleProtectedCache(access_key, catalog)) {
+      std::cout << "Protected cache Example failed" << std::endl;
+      return -1;
+    }
+  }
+  return 0;
+}
+
+int main(int argc, char** argv) {
+  static struct option long_options[] = {
+      {"example", required_argument, 0, 'e'},
+      {"key_id", required_argument, 0, 'i'},
+      {"key_secret", required_argument, 0, 's'},
+      {"catalog", required_argument, 0, 'c'},
+      {"layer_id", required_argument, 0, 'l'},
+      {"all", no_argument, 0, 'a'},
+      {"help", no_argument, 0, 'h'},
+      {0, 0, 0, 0}};
+
+  AccessKey access_key{};  // You can specify your here.access.key.id 
+                           // and here.access.key.secret
+  std::string catalog;     // the HRN of the catalog to which you to publish data
+  std::string layer_id;    // the of the layer inside the catalog to which you
+                           // want to publish data
```

Comment:
Redundant code comments? It's easy to find the meaning of these temporaries by the names, isn't it?

Reply:
User can set this parameters via command line or directly in code. With the comments it could be easier to find this parameters.

Reply:
I see. It makes sense then.